File: backend/src/main/java/com/project2/domain/post/dto/PostResponseDTO.java
package com.project2.domain.post.dto;

import java.util.List;

import com.project2.domain.member.dto.AuthorDTO;
import com.project2.domain.place.dto.PlaceDTO;
import com.project2.domain.post.entity.Post;
import com.project2.domain.post.entity.PostImage;
import com.project2.global.security.SecurityUser;

import lombok.Getter;

@Getter
public class PostResponseDTO {
	private final Long id;
	private final String title;
	private final String content;
	private final PlaceDTO placeDTO;
	private final Integer likeCount;
	private final Boolean isLiked;
	private final Integer scrapCount;
	private final Boolean isScrapped;
	private final Integer commentCount;
	private final List<String> imageUrls;
	private final AuthorDTO author;

	public PostResponseDTO(Post post, SecurityUser actor) {
		this.id = post.getId();
		this.title = post.getTitle();
		this.content = post.getContent();
		this.placeDTO = new PlaceDTO(post.getPlace().getName(), post.getPlace().getKrCategory());
		this.likeCount = post.getLikes().size();
		this.isLiked = post.getLikes().stream().anyMatch(like -> like.getMember().getId().equals(actor.getId()));
		this.scrapCount = post.getScraps().size();
		this.isScrapped = post.getScraps().stream().anyMatch(scrap -> scrap.getMember().getId().equals(actor.getId()));
		this.commentCount = post.getComments().size();
		this.imageUrls = post.getImages().stream().map(PostImage::getImageUrl).sorted().toList();
		this.author = new AuthorDTO(post.getMember().getId(), post.getMember().getNickname(),
			post.getMember().getProfileImageUrl());
	}
}

File: backend/src/main/kotlin/com/project2/domain/post/dto/PostResponseDTO.kt
package com.project2.domain.post.dto

import com.project2.domain.member.dto.AuthorDTO
import com.project2.domain.place.dto.PlaceDTO
import com.project2.domain.post.entity.Post
import com.project2.global.security.SecurityUser

class PostResponseDTO(
        post: Post,
        actor: SecurityUser
) {
    val id: Long = post.id!!
    val title: String = post.title
    val content: String = post.content
    val placeDTO: PlaceDTO = PlaceDTO(post.place.name, post.place.krCategory)
    val likeCount: Int = post.likes.size
    val isLiked: Boolean = post.likes.any { it.member.id == actor.id }
    val scrapCount: Int = post.scraps.size
    val isScrapped: Boolean = post.scraps.any { it.member.id == actor.id }
    val commentCount: Int = post.comments.size
    val imageUrls: List<String> = post.images.map { it.imageUrl }.sorted()
    val author: AuthorDTO = AuthorDTO(
            post.member.id!!,
            post.member.nickname,
            post.member.profileImageUrl
    )
}